### PR TITLE
BE implement new scoring system (all countries view only)

### PR DESCRIPTION
**Pull request by nastasi-oq**
With this PR we provide a new data structure for all countries, included score and rank.
It provide data for #305

### CODE REVIEW

**Comment by pzwsk** [comment]
Thanks @nastasi-oq 

I understand removing score for dataset and implementing decision tree will be done later then?

**Comment by nastasi-oq** [comment]
@pzwsk 
> I understand removing score for dataset

Not necessarily, we can add it in this PR

> and implementing decision tree will be done later then?

It is already there with help of scoring currently (but is trivial change it to be pure decision tree)

**Comment by nastasi-oq** [comment]
@pzwsk scoring, ranking and sorting disabled, ``dev.`` that seems to point to ``exp.`` Backend at this moment, seems not implement sorting yet, is it correct ?

**Comment by pzwsk** [comment]
Hi, as discussed, to me you can merge but we need to get rid of calls to api/country_scoring/ frontend side to be perfectly aligned with new indicators.

I am going to open a new issue for that.

**Comment by oncletom** [comment]
`api/country_scoring` is in use in the `dataset_list.html` page, and it also returns the list of datasets.

**Comment by oncletom** [comment]
@nastasi-oq is this route https://exp.riskopendata.org/api/scoring/AU supposed to output results? Because at the moment it throws an error but I'm not sure if it's because i'm not supposed to call it this way.

File: backend/ordd_api/models.py [hunk 1]
```diff
@@ -346,11 +349,29 @@ def gem_score_calculate(cls, inst):
...
+            cat = 2
+
+        return cat
+
     def save(self, *args, **kwargs):
         # to calculate scores, using m2m relationships, save a first
         # time the dataset before calculations is required
         super().save(*args, **kwargs)
         self.score, self.score_th_norm = self.gem_score_calculate(self)
+        self.score_new = 0
```
Comment (daniviga) [inline backend/ordd_api/models.py]: Please add a small comment about the fact that this is hardcoded to zero for now.

File: backend/ordd_api/views.py [hunk 2]
```diff
@@ -1789,6 +1858,16 @@ def get(self, request):
...
 
+class CountryScoringNewWorldGet(APIView):
+    """This view return the list of country with dataset instances and
+ their scores"""
+
+    def get(self, request):
+        ret = Score_new.all_country_scoring_new(request)
+        return Response(ret)
+
+    # select country_id, count(pri.part) as kd_num from (select country_id, keydataset_id as part from ordd_api_dataset group by country_id, keydataset_id) as pri GROUP BY country_id ORDER BY kd_num;
```
Comment (daniviga) [inline backend/ordd_api/views.py]: Stale comment to be removed

**Comment by nastasi-oq** [comment]
> @nastasi-oq is this route https://exp.riskopendata.org/api/scoring/AU supposed to output results? Because at the moment it throws an error but I'm not sure if it's because i'm not supposed to call it this way.

@oncletom no, I want to manage it in a separate PR, you must continue to use the old ``/api/country_scoring/<country_id>`` address.

**Pull request merged**